File: src/Laundro/Laundro.Core/Domain/Entities/Store.cs
namespace Laundro.Core.Domain.Entities;
public class Store : Entity
{
    public string? Name { get; set; }
    public string? Location { get; set; }

    public int TenantId { get; set; }
    public Tenant? Tenant { get; set; }

    public ICollection<StoreImage>? Images { get; set; }

    // These should not be nullable
    // To avoid this error: CS8620 - Argument cannot be used for parameter due to differences in the nullability of reference types.
    public ICollection<StoreUser> StoreUser { get; set; } = new List<StoreUser>();
}

File: src/client-app/src/models/store.ts
import StoreImage from './storeImage';
import StoreUser from './storeUser';
import Tenant from './tenant';

interface Store {
  id: number;
  createdAt?: string;
  name: string;
  location: string;
  tenantId: number;
  tenant: Tenant;
  storeUser?: StoreUser[];
  images: StoreImage[];
  obfuscatedId?: string;
}

export default Store;
